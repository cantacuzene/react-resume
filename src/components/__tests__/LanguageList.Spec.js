import React from 'react'
import LanguageList from '../LanguagesList'
import renderer from 'react-test-renderer'
//import HomePage from '../HomePage';

describe('LanguageList', ()=>{

    const articles ={
        "Languages":[{
        "name":"English",
        "rating":0.87,
        "id":0
    },
    {
        "name":"French",
        "rating":1,
        "id":1
    }],
    "chartid":"myd3chart" ,
    "width":500, 
    "height":190,
    "Titles":{"Languages":'Languages'}
}

    it('renders correctly', ()=>{
       const tree=  renderer.create(
        <LanguageList {...articles} />
        ).toJSON();
        expect(tree).toMatchSnapshot();

        //looking for the g elements
        expect(tree.children[1].children[0].children[0].children.length).toBe(2);
    })


});
