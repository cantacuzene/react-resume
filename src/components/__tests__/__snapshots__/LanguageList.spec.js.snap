// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageList renders correctly 1`] = `
<section
  className="sub lang"
>
  <h2>
    Languages
  </h2>
  <div
    className="center"
  >
    <div
      className="svgContainerWebChart"
      id="myd3chart"
    >
      <svg
        className="svg-content-responsive"
        preserveAspectRatio="xMinYMax meet"
        viewBox="0 0 500 190"
      >
        <g>
          <defs>
            <filter
              id="inset-shadow1"
            >
              <feOffset
                dx="0"
                dy="0"
              />
              <feGaussianBlur
                result="offset-blur"
                stdDeviation="5"
              />
              <feComposite
                in="SourceGraphic"
                in2="offset-blur"
                operator="out"
                result="inverse"
              />
              <feFlood
                floodColor="grey"
                floodOpacity="0.5"
                result="color"
              />
              <feComposite
                in="color"
                in2="inverse"
                operator="in"
                result="shadow"
              />
              <feComposite
                in="shadow"
                in2="SourceGraphic"
                operator="over"
              />
            </filter>
          </defs>
          <defs>
            <filter
              id="inset-shadow2"
            >
              <feOffset
                dx="0"
                dy="0"
              />
              <feGaussianBlur
                result="offset-blur"
                stdDeviation="1"
              />
              <feComposite
                in="SourceGraphic"
                in2="offset-blur"
                operator="out"
                result="inverse"
              />
              <feFlood
                floodColor="white"
                floodOpacity="0.5"
                result="color"
              />
              <feComposite
                in="color"
                in2="inverse"
                operator="in"
                result="shadow"
              />
              <feComposite
                in="shadow"
                in2="SourceGraphic"
                operator="over"
              />
            </filter>
          </defs>
          <path
            d="M5.204748896376251e-15,-85A85,85,0,1,1,-5.204748896376251e-15,85A85,85,0,1,1,5.204748896376251e-15,-85M-1.1940306291686693e-14,-65A65,65,0,1,0,1.1940306291686693e-14,65A65,65,0,1,0,-1.1940306291686693e-14,-65Z"
            fill="#ccc"
            style={
              Object {
                "filter": "url(#inset-shadow1)",
              }
            }
          />
          <path
            d="M-3.714968831558859,-74.23745016216914A10,10,0,0,1,7.108871608709576,-84.70220743552605A85,85,0,1,1,-69.16728766987411,-49.40532680180226A10,10,0,0,1,-54.18448864942558,-50.88262168756904L-54.18448864942558,-50.88262168756904A10,10,0,0,1,-52.89263174755081,-37.780544024907606A65,65,0,1,0,5.43619593607203,-64.7722762742258A10,10,0,0,1,-3.7149688315588607,-74.23745016216914Z"
            fill="#67BAF5"
            style={
              Object {
                "filter": "url(#inset-shadow2)",
              }
            }
          />
          <circle
            cx="0"
            cy="0"
            fill="#fff"
            fillOpacity="1"
            r={65}
          />
          <text
            dx="5"
            dy="15"
            fill={
              Rgb {
                "b": 204,
                "g": 204,
                "opacity": 1,
                "r": 204,
              }
            }
            style={
              Object {
                "fontSize": "35px",
              }
            }
            textAnchor="middle"
          >
            <tspan
              dy="0"
              x="0"
            >
              English
            </tspan>
          </text>
        </g>
        <g>
          <defs>
            <filter
              id="inset-shadow1"
            >
              <feOffset
                dx="0"
                dy="0"
              />
              <feGaussianBlur
                result="offset-blur"
                stdDeviation="5"
              />
              <feComposite
                in="SourceGraphic"
                in2="offset-blur"
                operator="out"
                result="inverse"
              />
              <feFlood
                floodColor="grey"
                floodOpacity="0.5"
                result="color"
              />
              <feComposite
                in="color"
                in2="inverse"
                operator="in"
                result="shadow"
              />
              <feComposite
                in="shadow"
                in2="SourceGraphic"
                operator="over"
              />
            </filter>
          </defs>
          <defs>
            <filter
              id="inset-shadow2"
            >
              <feOffset
                dx="0"
                dy="0"
              />
              <feGaussianBlur
                result="offset-blur"
                stdDeviation="1"
              />
              <feComposite
                in="SourceGraphic"
                in2="offset-blur"
                operator="out"
                result="inverse"
              />
              <feFlood
                floodColor="white"
                floodOpacity="0.5"
                result="color"
              />
              <feComposite
                in="color"
                in2="inverse"
                operator="in"
                result="shadow"
              />
              <feComposite
                in="shadow"
                in2="SourceGraphic"
                operator="over"
              />
            </filter>
          </defs>
          <path
            d="M5.204748896376251e-15,-85A85,85,0,1,1,-5.204748896376251e-15,85A85,85,0,1,1,5.204748896376251e-15,-85M-1.1940306291686693e-14,-65A65,65,0,1,0,1.1940306291686693e-14,65A65,65,0,1,0,-1.1940306291686693e-14,-65Z"
            fill="#ccc"
            style={
              Object {
                "filter": "url(#inset-shadow1)",
              }
            }
          />
          <path
            d="M-4.2482293880076565,-84.89377213357213A85,85,0,1,1,4.2482293880076565,84.89377213357213A85,85,0,1,1,-4.2482293880076565,-84.89377213357213M-1.1940306291686693e-14,-65A65,65,0,1,0,1.1940306291686693e-14,65A65,65,0,1,0,-1.1940306291686693e-14,-65Z"
            fill="#67BAF5"
            style={
              Object {
                "filter": "url(#inset-shadow2)",
              }
            }
          />
          <circle
            cx="0"
            cy="0"
            fill="#fff"
            fillOpacity="1"
            r={65}
          />
          <text
            dx="5"
            dy="15"
            fill={
              Rgb {
                "b": 204,
                "g": 204,
                "opacity": 1,
                "r": 204,
              }
            }
            style={
              Object {
                "fontSize": "35px",
              }
            }
            textAnchor="middle"
          >
            <tspan
              dy="0"
              x="0"
            >
              French
            </tspan>
          </text>
        </g>
      </svg>
    </div>
  </div>
</section>
`;
